### PR TITLE
Track number of steps on lava and add a default negative reward for each step

### PR DESCRIPTION
**Pull request by jaudick**
[UNITY](https://github.com/NextCenturyCorporation/ai2thor/pull/188)

### CODE REVIEW

**Review by kdrumm — COMMENTED** [review]
I am getting unit test failures.  I think they are due to the mock_controller needing to be updated.

File: machine_common_sense/reward.py [hunk 1]
```diff
@@ -9,6 +9,7 @@
 GOAL_ACHIEVED = 1
 GOAL_NOT_ACHIEVED = 0
 STEP_PENALTY = 0.001
+LAVA_PENTALTY = 1000
```
Comment (ThomasSchellenbergNextCentury) [inline machine_common_sense/reward.py:12]: Starting a discussion on this number:

1. I'm trying to remember how the reward code actually works. I believe you start at `1`, and then lose `0.001` each step, so after 1000 steps, your reward is negative.
2. If that's the case, then the LAVA_PENALTY should be between 0 and 1.
3. From last week's engineering collab meeting, we decided that a single step in the lava should be allowed. I took that concept to mean that the reward would still be positive after a single step in the lava. If we're in agreement, I propose that the LAVA_PENALTY be `0.5`. This means that a single step in the lava doesn't make the reward negative, but two steps does.

Thoughts?

File: machine_common_sense/reward.py [hunk 2]
```diff
@@ -247,4 +251,5 @@ def calculate_reward(
                                                                      objects,
                                                                      agent,
                                                                      reach)
-        return Reward._adjust_score_penalty(current_score, number_steps)
+        return Reward._adjust_score_penalty(
+            current_score, number_steps, steps_on_lava)
```
Comment (ThomasSchellenbergNextCentury) [inline machine_common_sense/reward.py:255]: We should add some unit tests for the new behavior.

File: machine_common_sense/controller_logger.py [hunk 3]
```diff
@@ -53,6 +53,7 @@ def _write_debug_output(self, payload):
         logger.debug("  POSITION: " + str(step_output.position))
         logger.debug("  ROTATION: " + str(step_output.rotation))
         logger.debug("  HAPTIC FEEDBACK: " + str(step_output.haptic_feedback))
+        logger.debug("  STEPS ON LAVA: " + str(step_output.steps_on_lava))
```
Comment (ThomasSchellenbergNextCentury) [inline machine_common_sense/controller_logger.py:56]: I don't think this property needs to be in the debug output.

File: machine_common_sense/reward.py (hunk 1 above)
Comment (deanwetherby) [reply machine_common_sense/reward.py:12]: Seconded. 0.5 is a large enough penalty compared to 0.001 and we will provide an override in the configuration anyway.

File: machine_common_sense/reward.py (hunk 2 above)
Comment (jaudick) [reply machine_common_sense/reward.py:255]: I think unit tests should be a separate ticket, what do you think?

Comment (deanwetherby) [reply machine_common_sense/reward.py:255]: Unit tests go hand-in-hand with the ticket and code modifications. You may, like I frequently do, discover an issue while testing.

**Comment by jaudick** [comment]
@deanwetherby tests look good and pass for me

File: machine_common_sense/controller_logger.py (hunk 3 above)
Comment (ThomasSchellenbergNextCentury) [reply machine_common_sense/controller_logger.py:56]: Bump

Comment (jaudick) [reply machine_common_sense/controller_logger.py]: Fixed

**Review by rartiss55 — APPROVED** [review]

**Review by ThomasSchellenbergNextCentury — APPROVED** [review]

**Pull request merged**